backend/utils/llm.py

```diff
@@ -24,5 +25,28 @@ def call_model(system_prompt, user_prompt):
       ],
       max_tokens=125
    )
-   logger.info("{0} response : \"{1}\"".format(config.mlarge_model, response.choices[0].message.content))
+   logger.info("{0} response : \"{1}\"".format(config.mistral_model, response.choices[0].message.content))
    return response.choices[0].message.content
+
+def call_model_with_tools(system_prompt, user_prompt, tools):
+   logger.info("called llm with tools. Waiting on response model with prompt {0}".format(str([system_prompt, user_prompt])))
+   response = client.chat(
+      model=config.mistral_model, 
+      messages=[
+         ChatMessage(
+            role="system",
+            content=system_prompt
+         ),
+         ChatMessage(
+            role="user",
+            content=user_prompt
+         )
+      ], 
+      tools=tools, 
+      tool_choice="any"
+   )
+   logger.info("{0} response : \"{1}\"".format(config.mistral_model, response.choices[0].message.content))
```

Comment:
a lot of this is repeated. I wonder if it would be neater to split out into several methods. I'm thinking `getResponse` , `call_model` and `call_model_with_tools` and then we'd only need to expose `call_model` and `call_model_with_tools` . (this would only work is tools can be defaulted to `None` in the `getResponse` method)